TarkovMapOverlay/MainWindow.xaml.cs

```diff
@@ -201,5 +324,130 @@ private void MinimizeKeybind_OnClick(object sender, RoutedEventArgs e)
+        }
+
+        private void EnableMapListIfNotEmpty()
+        {
+            if (List_CustomMaps.Items.Count != 0)
+            {
+                List_CustomMaps.IsEnabled = true;
+            }
+        }
+
+        private bool SaveSettings() 
+        {
+            if (!Directory.Exists("Settings"))
+            {
+                Directory.CreateDirectory("Settings");
+            }
+
+            SavedSettings settings = new SavedSettings();
+            settings.windowTop = this.Top;
+            settings.windowLeft = this.Left;
+            settings.windowHeight = this.Height;
+            settings.windowWidth = this.Width;
+
+            settings.visual_opacity = this.Opacity;
+            settings.visual_transparency = transparentBackground;
+
+            if (TarkovMap.Source != null)
+            {
+                settings.currentMapPath = TarkovMap.Source.ToString();
+            }
+            else {
+                settings.currentMapPath = null;
+            }
+
+            settings.customMapList.AddRange(_savedMaps);
+            settings.minimizeKey = minimizeKey;
+            settings.minimizeMousebutton = minimizeButton;
+
+            BinaryFormatter bf = new BinaryFormatter();
+            FileStream fs = new FileStream("Settings/settings.sav", FileMode.Create);
+            bf.Serialize(fs, settings);
+            fs.Close();
+            return true;
+        }
+
+        private SavedSettings LoadSettings()
+        {
+            if (File.Exists("Settings/settings.sav")) {
+                FileStream fs = new FileStream("Settings/settings.sav", FileMode.Open);
+                BinaryFormatter bf = new BinaryFormatter();
+
+                SavedSettings settings = (SavedSettings)bf.Deserialize(fs);
+
+                return settings;
+            }
+
+            return new SavedSettings();
+        }
+
+        public void MoveIntoView() //important if saved WindowPosition is out of screenbounds ( if saved on 2.screen for example )
+        {
+            double _windowTop = this.Top;
+            double _windowHeight = this.Height;
+            double _windowLeft = this.Left;
+            double _windowWidth = this.Width;
+            
+            if (_windowTop + _windowHeight / 2 > SystemParameters.VirtualScreenHeight)
+            {
+                _windowTop = SystemParameters.VirtualScreenHeight -_windowHeight;
+            }
+
+            if (_windowLeft + _windowWidth / 2 > SystemParameters.VirtualScreenWidth)
+            {
+                _windowLeft = SystemParameters.VirtualScreenWidth - _windowWidth;
+            }
+
+            if (_windowTop < 0)
+            {
+                _windowTop = 0;
+            }
+
+            if (_windowLeft < 0)
+            {
+                _windowLeft = 0;
+            }
+
+            this.Top = _windowTop;
+            this.Height = _windowHeight;
+            this.Left = _windowLeft;
+            this.Width = _windowWidth;
+        }
```

Comment:
For some reason the window opens with a 0 by 0 size if there are no settings saved previously and it's very hard to find 😄 

Also, it may be good if the window doesn't start in the top left of the screen if there are no settings set.

Reply:
i think this relates to the fact that the SavedSettings class is missing somehow. XD ( 0x0 size )
do you have a screenshot from the Windows startposition? Normally it should be at 200,200 if it's saved out of screen when you had a second monitor and now just one.

Reply:
it was in the very top left corner. I think it may have been because I had a save file already from previous testing.

Reply:
I cannot reproduce this behaviour. maybe just because the SavedSettings - class was missed the first time, and it defaults to 0 in this case.